src/classes/app.py

```diff
@@ -203,11 +179,77 @@ def __init__(self, *args, mode=None):
         # Track which dockable window received a context menu
         self.context_menu_object = None
 
+        if info.LAUNCH_MODE == "gui":
+            self.configure_gui()
+
+        # Create main window
+        from windows.main_window import MainWindow
+        self.window = MainWindow()
+
+        # Reset undo/redo history
+        self.updates.reset()
+        self.window.updateStatusChanged(False, False)
+
+        # Connect our exit signals
+        self.aboutToQuit.connect(self.cleanup)
+
+        log.info('Process command-line arguments: %s' % args)
+        if len(args[0]) == 2:
+            path = args[0][1]
+            if ".osp" in path:
+                # Auto load project passed as argument
+                self.window.OpenProjectSignal.emit(path)
+            else:
+                # Apply the default settings and Auto import media file
+                self.project.load("")
+                self.window.filesView.add_file(path)
+        else:
+            if info.LAUNCH_MODE != "unittest":
+                # Recover backup file (this can't happen until after the Main Window has completely loaded)
+                self.window.RecoverBackup.emit()
+
+    def settings_load_error(self, filepath=None):
+        """Use QMessageBox to warn the user of a settings load issue"""
+        _ = self._tr
+        from classes import info
+        if info.LAUNCH_MODE == "gui":
+            QMessageBox.warning(
+                None,
+                _("Settings Error"),
+                _("Error loading settings file: %(file_path)s. Settings will be reset.")
+                % {"file_path": filepath}
+            )
+
+    def _tr(self, message):
+        return self.translate("", message)
+
+    # Start event loop
+    def run(self):
+        """ Start the primary Qt event loop for the interface """
+        return self.exec_()
+
+    @pyqtSlot()
+    def cleanup(self):
+        """aboutToQuit signal handler for application exit"""
+        try:
+            from classes.logger import log
+            self.settings.save()
+        except Exception:
+            log.error("Couldn't save user settings on exit.", exc_info=1)
+
+    def configure_gui(self):
+        """Apply application GUI settings for fonts and colors"""
+        from classes import info, ui_util
+        from classes.logger import log
+
+        # Load ui theme if not set by OS
+        ui_util.load_theme(self.settings)
+
         # Set Font for any theme
         if self.settings.get("theme") != "No Theme":
             # Load embedded font
             try:
-                log.info("Setting font to %s" % os.path.join(info.IMAGES_PATH, "fonts", "Ubuntu-R.ttf"))
+                log.info("Setting font to %s", os.path.join(info.IMAGES_PATH, "fonts", "Ubuntu-R.ttf"))
```

Comment:
Does this still need the %s placeholder?

Reply:
It does. logging's formatter processes template strings with `%` escapes, the remaining arguments are the inputs to the formatting.

There's one key difference between passing the format string and inputs as arguments, and doing the string interpolation up front: logging calls benefit from lazy evaluation. If logging is disabled (or the `logging.INFO` level isn't configured for output), not only will the final, formatted string not be generated, but the `os.path.join()` won't even be executed, since its output isn't needed.

`"Setting font to %s" % os.path.join(info.IMAGES_PATH, "fonts", "Ubuntu-R.ttf")` always has to do the `os.path.join()` and generate the string, in order to pass it to the `log.info()` call — whether or not it will ultimately be logged.